src/components/TsJsBlock/index.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+
+import { format } from 'prettier/standalone'
+import { parsers } from 'prettier/parser-babel'
+import { transform } from '@babel/standalone'
+
+import Tabs from '@theme/Tabs'
+import TabItem from '@theme/TabItem'
+
+import CodeBlock from '@theme/CodeBlock'
+
+const TsJsBlock = ({ children, fileName, ...props }) => {
+  const tsSnippet = children
+  // 1. Transpile TS to JS
+  const { code: jsSnippet } = transform(tsSnippet, {
+    plugins: ['transform-typescript'],
+    retainLines: true,
+  })
+  // 2. Prettify the resulting JS
+  // FIXME: Maybe we can pass this as a config to the prop?
```

Comment:
You could use `useContext` for sending the config down from the root of the app, if you want

Reply:
Did that, thank you! https://github.com/KILTprotocol/docs/pull/147/commits/5863573ff78e35cdcf483987c4b97837d7996d02

Did not find a way to open the `.prettierrc` file directly, so there is a bit of duplication. Do you think it's fine?

Reply:
I think the duplication is fine. I would suggest putting the prettierConfig into the customFields as a separate property though, in  case want to add more custom fields in the future
```
  customFields: {
    prettierConfig: {
      trailingComma: "es5",
      semi: false,
      singleQuote: true,
      printWidth: 80
    }
  }
```